packages/fetchai/connections/p2p_libp2p_client/connection.py

```diff
@@ -235,21 +229,21 @@ async def _perform_connection_to_node(self) -> None:
 
     async def _setup_connection(self) -> None:
         """Set up connection to node over tcp connection."""
-        record = AgentRecordPb()
-        record.address = self.node_por.address
-        record.public_key = self.node_por.public_key
-        record.peer_public_key = self.node_por.representative_public_key
-        record.signature = self.node_por.signature
-        record.service_id = POR_DEFAULT_SERVICE_ID
-        record.ledger_id = self.node_por.ledger_id
-
-        registration = Register()
-        registration.record.CopyFrom(record)  # pylint: disable=no-member
-        msg = AcnMessage()
-        msg.version = ACN_CURRENT_VERSION
-        msg.register.CopyFrom(registration)  # pylint: disable=no-member
-
-        buf = msg.SerializeToString()
+        agent_record = AcnMessage.AgentRecord(
+            address=self.node_por.address,
+            public_key=self.node_por.public_key,
+            peer_public_key=self.node_por.representative_public_key,
+            signature=self.node_por.signature,
+            service_id=POR_DEFAULT_SERVICE_ID,
+            ledger_id=self.node_por.ledger_id,
+        )
+
+        acn_msg = acn_pb2.AcnMessage()
```

Comment:
I think it makes sense. The dialogue system was designed for agent-to-agent communication, but it doesn't refer explicitly to agent constructs or classes. As you noted, it looks like it is useful in other contexts too, e.g. in connections. So I am favourable to this approach. 

In dialogue modules, we should remove references to 'agent' concepts (e.g. in docstrings) and in the documentation, stress the fact that it is an agent independent dialogue system.

Reply:
sure, we can use acn protocol just like it done for other protocols for agents, so create own dialogues instance to handle state of the acn communication.
it can be done as independent handler that populates incoming envelope queue to be transfered to the network

Reply:
I think we should not do this in this PR. The PR is already large